tslearn/metrics/soft_dtw_fast.py

```diff
@@ -0,0 +1,94 @@
+# Author: Mathieu Blondel
+# License: Simplified BSD
+
+# encoding: utf-8
+
+import numpy as np
+from numba import njit, prange
+
+DTYPE = np.float64
+DBL_MAX = np.finfo("double").max
+
+
+@njit(parallel=True)
+def _softmin3(a, b, c, gamma):
+    a /= -gamma
+    b /= -gamma
+    c /= -gamma
+
+    max_val = max(max(a, b), c)
+
+    tmp = 0
+    tmp += np.exp(a - max_val)
+    tmp += np.exp(b - max_val)
+    tmp += np.exp(c - max_val)
+
+    return -gamma * (np.log(tmp) + max_val)
+
+
+@njit(parallel=True)
+def _soft_dtw(D, R, gamma):
+
+    m = D.shape[0]
+    n = D.shape[1]
+
+    # Initialization.
+    for i in prange(m + 1):
+        R[i, 0] = DBL_MAX
+
+    for j in prange(n + 1):
+        R[0, j] = DBL_MAX
+
```

Comment:
For the matrix R, I think that it only the first row (from index 1 to n) and the first column (from index 1 to m) should be equal to DBL_MAX and the coefficient of index (0, 0) should be equal to zero. The other coefficients are computed in the loops after.
About DBL_MAX, I chose to use of np.finfo('double').max because it has a dtype:
- np.finfo('double').max.dtype returns dtype('float64')
- np.inf.dtype returns AttributeError: 'float' object has no attribute 'dtype'

Reply:
Ahh...right.. so I should have written R[0, ...]